pywbem/_cim_types.py

```diff
@@ -655,7 +661,14 @@ def get_local_utcoffset():
         A positive value indicates minutes east of UTC, and a negative
         value indicates minutes west of UTC.
         """
-        utc = datetime.utcnow()
+        # Effective Python 3.12, utcnow must define tz to avoid warning msg
+        # That is because utcnow returns a niave datetime object if no timezone
+        # is provided to the function
+        if sys.version_info[0:2] <= (3, 12):
+            utc = datetime.now(timezone.utc)
+        else:
+            utc = datetime.utcnow()
```

Comment:
`now(tz)` returns a timezone-aware object, while `now()` returns a timezone-naive object.
This will cause the comparison `local < utc` further down to fail because they cannot be compared.

I suggest instead to replace the whole calculation in this method with:
```
return -time-ztimezone
```
See https://docs.python.org/3/library/time.html#time.timezone
Note that `time.timezone` does not consider DST, but then I'm not sure what the DST rules are for `get_local_utcoffset()`.

Reply:
it was late in the day.

Reply:
The division by 60 only converts seconds to minutes, so it assumes that timezones are on full minutes. The old algorithm made the same assumption.